server/container/guice/guice-common/src/test/java/org/apache/james/utils/InMemoryMailRepositoryStoreTest.java

```diff
@@ -257,4 +266,11 @@ public void selectShouldNotReturnDifferentResultsWhenUsedInAConcurrentEnvironmen
         assertThat(actualSize).isEqualTo(threadCount);
     }
 
+    @Test
+    public void selectShouldNotAddUrlWhenProtocolDoNotExist() {
+        assertThatThrownBy(() -> repositoryStore.select(UNKNOWN_PROTOCOL_REPO));
```

Comment:
should have one more assertion `.isInstanceOf()`?

Reply:
Well, the assertion here is "that the URL is not registered".

Not that we throw an exception (there is another test for that).

The question is "how do we ignore an exception". I found the **assertThatThrownBy** being better than a **try/catch block**.